### DISCUSSION TITLE
Fetch Very Last Item From Base

### DISCUSSION THREAD

**Discussion by HanzCEO** [deta/cloud-docs#243] [Ideas] [answered]
Hi, i like DetaBase

But i feel guilty and inefficient when i want to fetch the latest pushed item from Base.
I know i can fastforward until `res.last` is falsy but this thing is worrying me.

I think adding `db.fetchLast` wouldn't hurt.
What do you think guys?

**Comment by abdelhai** [comment, marked as answer]
hi, thanks!
in base, there's nothing called "first" and "last". you can optimize the items you are storing a bit and maybe use a datetime sorted key and look for a key that was created in the last x hours/minutes/etc and then narrow down the last on the last item. this should make queries efficient and fast.

**Reply by HanzCEO** [reply]
Thanks for the inspiration!

**Reply by fanfanmyid** [reply]
Hi, can you give me some example to do this?

**Reply by abdelhai** [reply]
check out this answer https://github.com/orgs/deta/discussions/344#discussioncomment-3263320
